Update categories.css

izaqueIsrael opened the pull request:
Tentei dar um query selector para colocar as classes 'categories__item, categories__icon e categories__text' no evento click, porém só funcionava para o primeiro elemento, vi que para consertar isso, teria que usar querySelectorAll, criar um seletor com forEach, iria ser um trampo enorme e como no momento não domino JS e o pedido foi para que não o usasse, deixei essas estilizações no hover mesmo que não seja o pedido no figma

izaqueIsrael committed:
Update categories.css

Aditions: class styling navbar navigation, read the comentary

thiagopuntar commented on src/styles/categories.css:
```diff
@@ -0,0 +1,36 @@
+}
+
+.categories__icon:hover{
+  filter: brightness(0) saturate(100%) invert(100%) sepia(0%) saturate(1941%) hue-rotate(238deg) brightness(117%) contrast(100%);
```
Bem interessante a forma que você encontrou para resolver isso... Eu estava pensando em algo mais voltado a esse artigo https://css-tricks.com/change-color-of-svg-on-hover/. Depois me manda a referência que você encontrou pra resolver com filter. 

Só ficou algo um pouco mais verboso e de repente no longo prazo fica mais difícil de dar manutenção.

Aqui, também acho que rola de trabalhar com seletor composto, porque o ícone só muda de cor quando de ato a gente faz o hover nele, o comportamento padrão seria fazer o cover no item como um todo e tanto ícone quanto text mudarem de cor.

izaqueIsrael replied:
A ferramenta foi essa aqui, peguei em algum comentário no stackoverflow, é só colocar a cor que deseja para o svg que é sucesso https://codepen.io/sosuke/pen/Pjoqqp

thiagopuntar commented on src/styles/categories.css:
```diff
@@ -0,0 +1,36 @@
+}
+
+.categories__text:hover{
+  font-size: 1.5rem;
```
A respeito do font-size e do width, você gostaria de aumentar o tamanho do ícone? O melhor é usar a propriedade transform. Quando a gente altera o width ou font-size, acaba deslocando os outros elementos da página e não fica um efeito legal. Dá uma pesquisada no scale do transform.

thiagopuntar reviewed (changes requested):
Fala Izaque, muito obrigado por sua PR, parabéns por ter conseguido resolver o problema. Fiz alguns comentários a respeito da implementação, qualquer dúvida me fala.

Um detalhe bem importante, a branch de base tem que ser a cohort2, vi que você abriu para a main.

Outra coisa importante é adicionar transition para melhorar o efeito do hover.

izaqueIsrael committed:
navbar fix

Agora ela funciona como deveria funcionar.

*Alterei o font-size para transform como o pedido.

*Adicionei o código JS para limpar as classes da navbar e depois adicionar para o elemento clicado

isaac-israel commented:
Outra coisa que poderia fazer seria adicionar um transitions no hover, categories__item e os 2 actived, não fiz porque não sei como querem no figma

thiagopuntar commented:
Fala Izaque, blz? Bom dia! De fato está tendo conflitos aqui na sua branch. Vamos fazer o seguinte, eu acho que vai ser um bom case pros colegas também, porque ter conflito na branch é algo bem normal do dia a dia. Então vou fazer o rebase ao vivo e atualizar sua branch, aí a partir disso a gente revisa, pode ser?
Feliz ano novo meu amigo... tudo de bom e muito obrigado pela sua participação de sempre. 😄

izaqueIsrael committed:
o git mandou commitar

isaac-israel commented:
o vs code não apita mais nada, creio que resolvi o problema

izaqueIsrael committed:
Merge branch 'cohort2' into nav-icons

izaqueIsrael committed:
adicionando animações

izaqueIsrael committed:
atualizando a animação

izaqueIsrael committed:
atualizei o background, ficou mais parecido

izaqueIsrael committed:
alterações que ia fazer e reverti

izaqueIsrael committed:
modal do sucesso

izaqueIsrael committed:
estilizando o modal

izaqueIsrael committed:
Merge branch 'cohort2' into nav-icons

thiagopuntar commented on src/styles/modal.css:
```diff
@@ -21,3 +21,28 @@
+  font-family: "Inter", Arial, sans-serif;
+  font-style: bold;
+  font-weight: 900;
+  font-size: 24px;
```
font-size em rem

thiagopuntar commented on src/styles/modal.css:
```diff
@@ -21,3 +21,28 @@
+  font-style: bold;
+  font-weight: 900;
+  font-size: 24px;
+  line-height: 21px;
```
line-height em proporção

thiagopuntar commented on src/styles/modal.css:
```diff
@@ -21,3 +21,28 @@
+}
+
+.modal__text{
+  font-family: 'Lato', Arial, sans-serif;
```
Como ponto de melhoria, tenta utilizar uma variável do css para a gente isolar isso

thiagopuntar commented on src/styles/modal.css:
```diff
@@ -21,3 +21,28 @@
+  font-weight: 400;
+  font-size: 16px;
+  line-height: 19px;
+  color: #3A5667; 
```
Usar cor existente da variável do CSS, se não tiver criar uma nova (mas utilizar em HSL)

thiagopuntar commented on src/styles/modal.css:
```diff
@@ -21,3 +21,28 @@
+}
+
+.modal__title{
+  font-family: "Inter", Arial, sans-serif;
```
Eu acho que a gente setou a font inter como font global, não precisa dessa prop

thiagopuntar commented on src/styles/reset.css:
```diff
@@ -5,11 +5,3 @@
   box-sizing: border-box;
 }
-
-body {
```
Porque a gente está tirando o tema?

thiagopuntar commented on src/styles/index.css:
```diff
@@ -1,4 +1,6 @@
 @import url(./reset.css);
+@import url(./body.css);
+@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap');
```
A gente já está importando todas as fonts no index.html, pode remover esse import

thiagopuntar commented on src/styles/body.css:
```diff
@@ -0,0 +1,7 @@
+body {
```
Ah entendi, você moveu o que tava no reset pro body, pode deixar da forma que estava.

thiagopuntar commented on src/scripts/index.js:
```diff
@@ -82,6 +82,34 @@ buttons.forEach(function (button) {
+  
+  resetClasses();
+
+  navIcon.classList.add('categories__icon_actived');
```
Acho que você consegue o mesmo resultado aplicando somente uma classe no parent. Fica uma solução mais elegante. e aí só utilizar seletores compostos no CSS

izaqueIsrael replied:
Eu havia deixado separado porque no figma as escalas de crescimento para a fonte e para o ícone estão levemente diferentes

thiagopuntar commented on src/scripts/index.js:
```diff
@@ -82,6 +82,34 @@ buttons.forEach(function (button) {
+
+function resetClasses() {
+  const navIcons = document.querySelectorAll('.categories__icon');
+  navIcons.forEach((element) => element.className = 'categories__icon');
```
ok o reset, só ao invés de atribuir o className diretamente eu utilizaria o classList.remove para remover a classe de active do parent. Dessa forma a gente tem mais robustez ao longo do caminho, pois vai que a gente adiciona outra classe nesses elementos no futuro? Aí você teria que lembrar de alterar no JS também.

thiagopuntar reviewed (changes requested):
Boa Izaque, vi que de fato acabou por fazer duas tasks na mesma branch.
Muito cuidado nas próximas, isso é fonte de confusão (e super normal, colega experiente faz isso também), mas de qualquer forma, vou reforçar para sempre se concentrar quando for fazer uma nova branch.

O modal está certinho, só fiz algumas sugestões de alteração do CSS.

Quanto às categorias, o efeito é basicamente esse, fiz alguns comentários de ajuste do código mas em geral temos a funcionalidade. Eu apenas:
- Diminuiria o transition para uns 0.2s ou algo do tipo
- Usaria um scale um pouco menor, talvez 1.1, tem que fazer o teste, mas estou achando bem grande
- E precisamos desse mesmo estilo no :hover

Com essa revisão e esse pontos acima, acho que a gente consegue mergear.

Importante, como a branch principal andou novamente, é interessante que você faça o rebase da sua branch mais uma vez.

izaqueIsrael committed:
Alterações pedidas para o merge

izaqueIsrael committed:
Merge branch 'cohort2/main' into nav-icons

izaqueIsrael committed:
alterando cor para hsl

izaqueIsrael committed:
Merge branch 'nav-icons' of https://github.com/practicum-wd-brazil/pdv into nav-icons
